### DISCUSSION TITLE
Error in is.type(x) : object 'ffi_is_formula' not found

### DISCUSSION THREAD

**Discussion by graliuce** [lindeloev/mcp#150] [Q&A] [answered]
Thank you for creating this package and all the accompanying resources. I've installed JAGS and mcp, and I'm currently trying to run the sample code. However, I keep running into the error "Error in is.type(x) : object 'ffi_is_formula' not found" whenever calling any of the mcp functions. 

The code I'm running is below and the error happens one the mcp_example line.

```
library(mcp)

# Define the model
model = list(
  response ~ 1,  # plateau (int_1)
  ~ 0 + time,    # joined slope (time_2) at cp_1
  ~ 1 + time     # disjoined slope (int_3, time_3) at cp_2
)

# Get example data and fit it
ex = mcp_example("demo", sample = FALSE)
fit = mcp(model, data = ex$data)
```

Any suggestions or advice would be appreciated!

**Comment by lindeloev** [comment, marked as answer]
Another user wrote me with this issue and solved it by upgrading R and all packages to the newest versions. Perhaps it was caused by the `rlang` package version being incompatible with some others, but this remains to be verified. `ffi_is_formula` is in the `rlang` package which is a dependency of a dependency of `mcp` :-)

That users issue arose on R 4.1.2 with rlang 0.4.12. You can call `sessionInfo()` to see yours.

Let me know how it goes!

**Reply by graliuce** [reply]
That's really helpful, thank you so much!
